Blog.API/Blog.Infrastructure/MediatR/Handlers/Categories/CreateCategoryHandler.cs

```diff
@@ -1,29 +1,41 @@
-﻿using System.Threading;
-using System.Threading.Tasks;
-using Blog.Core.Entities;
+﻿using Blog.Core.Contracts.Controllers.Categories;
+using Blog.Core.Contracts.ResponseDtos;
 using Blog.Core.MediatR.Commands.Categories;
 using Blog.Infrastructure.Services.Interfaces;
-using MapsterMapper;
+using FluentValidation;
 using MediatR;
+using Microsoft.Extensions.Logging;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Blog.Infrastructure.MediatR.Handlers.Categories;
 
-public sealed class CreateCategoryHandler : IRequestHandler<CreateCategoryCommand, Category>
+public sealed class CreateCategoryHandler : IRequestHandler<CreateCategoryCommand, CategoryResponse?>
 {
     private readonly ICategoryService _categoryService;
-    private readonly IMapper _mapper;
+    private readonly IValidator<CreateCategoryRequest> _createCategoryValidator;
+    private readonly ILogger _logger;
 
-    public CreateCategoryHandler(ICategoryService categoryService, IMapper mapper)
+    public CreateCategoryHandler(ICategoryService categoryService,
+        IValidator<CreateCategoryRequest> createCategoryValidator,
+        ILogger<CreateCategoryHandler> logger)
     {
         _categoryService = categoryService;
-        _mapper = mapper;
+        _createCategoryValidator = createCategoryValidator;
+        _logger = logger;
     }
 
-    public async Task<Category> Handle(CreateCategoryCommand request, CancellationToken cancellationToken)
+    public async Task<CategoryResponse?> Handle(CreateCategoryCommand request, CancellationToken cancellationToken)
     {
-        var category = _mapper.Map<Category>(request.Category);
-        var responseCategory = await _categoryService.CreateCategory(category);
+        var result = _createCategoryValidator.Validate(request.Category);
+        if (!result.IsValid)
+        {
+            _logger.LogError("CreateCategoryRequest object errors:\n{errors}", result.Errors.Select(e => e.ErrorMessage));
```

Comment:
Variable in the template should have better naming, errors is not clear. In that case it should be something like
```suggestion
            _logger.LogError("CreateCategoryRequest object errors:\n{createCategoryErrors}", result.Errors.Select(e => e.ErrorMessage));
```